internal/network/manager.go

```diff
@@ -247,68 +266,75 @@ func (m *Manager) Close() error {
 	//    continue the chain the Mux has to be closed.
 
 	// Close SCTP. This should close the data channels, SCTP, and DTLS
-	var errSCTP, errMux error
+	var errSCTP, errMux, errSRTP, errSRTCP error
 	if m.sctpAssociation != nil {
 		errSCTP = m.sctpAssociation.Close()
 	}
 
+	if m.srtpSession != nil {
+		errSRTP = m.srtpSession.Close()
+	}
+
+	if m.srtcpSession != nil {
+		errSRTCP = m.srtcpSession.Close()
+	}
+
 	// Close the Mux. This should close the Mux and ICE.
 	if m.mux != nil {
 		errMux = m.mux.Close()
 	}
 
 	// TODO: better way to combine/handle errors?
 	if errSCTP != nil ||
-		errMux != nil {
-		return fmt.Errorf("Failed to close: %v, %v", errSCTP, errMux)
+		errMux != nil ||
+		errSRTP != nil ||
+		errSRTCP != nil {
+		return fmt.Errorf("Failed to close: %v, %v, %v, %v", errSCTP, errMux, errSRTP, errSRTCP)
 	}
 
 	return nil
 }
 
 // SendRTP finds a connected port and sends the passed RTP packet
 func (m *Manager) SendRTP(packet *rtp.Packet) {
-	m.srtpOutboundContextLock.Lock()
-	defer m.srtpOutboundContextLock.Unlock()
-	if m.srtpOutboundContext == nil {
+	if m.srtpSession == nil {
 		// TODO log-level
 		// fmt.Printf("Tried to send RTP packet but no SRTP Context to handle it \n")
 		return
 	}
 
-	if ok := m.srtpOutboundContext.EncryptRTP(packet); !ok {
-		fmt.Println("SendRTP failed to encrypt packet")
-		return
-	}
-
 	raw, err := packet.Marshal()
 	if err != nil {
 		fmt.Printf("SendRTP failed to marshal packet: %s \n", err.Error())
+		return
 	}
 
-	_, err = m.iceConn.Write(raw)
+	writeStream, err := m.srtpSession.OpenWriteStream()
```

Comment:
This feels weird. I would explicitly open a stream for each RTCTrack so you can directly write to it instead of implicitly opening it when writing to the network manager.

Reply:
Yep agree 100%!

Every `RTCTrack` can call `OpenWriteStream` and keep their stream private. They will all just share one (even if they think they each have their own)